src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -2422,16 +2422,31 @@ namespace ngraph
                     writer << "                         {" << join(out[0].get_shape()) << "});\n";
                 }
 #else
-                kernel::emit_replace_slice(writer,
-                                           args[0].get_element_type().c_type_string(),
-                                           args[0].get_name(),
-                                           args[1].get_name(),
-                                           out[0].get_name(),
-                                           args[1].get_shape(),
-                                           out[0].get_shape(),
-                                           replace_slice->get_lower_bounds(),
-                                           replace_slice->get_upper_bounds(),
-                                           replace_slice->get_strides());
+                if (args[0].get_name() != out[0].get_name())
+                {
+                    kernel::emit_replace_slice(writer,
```

Comment:
why can't we use the original version of `emit_replace_slice`?

Reply:
My main intention is to avoid a copy in emit_replace_slice(). 
So, I was self debating where to do the if check : inside `emit_replace_slice()` or even inside `emit_pointwise_copy()`
However, I thought it was easier to read and understand the intention from the very top. Hence i added there.
On second thoughts, `emit_replace_slice()` might be a good place too. I will change it that way.